client/components/data/query-embed/index.jsx

```diff
@@ -0,0 +1,36 @@
+/**
+ * External dependencies
+ */
+import PropTypes from 'prop-types';
+import { Component } from 'react';
+import { connect } from 'react-redux';
+
+/**
+ * Internal dependencies
+ */
+import { requestEmbed } from 'state/embeds/actions';
+
+class QueryEmbeds extends Component {
+	static propTypes = {
+		siteId: PropTypes.number.isRequired,
+		url: PropTypes.string.isRequired,
+	};
+
+	componentDidMount() {
+		this.props.requestEmbed( this.props.siteId, this.props.url );
+	}
+
+	componentDidUpdate( prevProps ) {
+		if ( prevProps.siteId !== this.props.siteId || prevProps.url !== this.props.url ) {
+			this.props.requestEmbed( this.props.siteId, this.props.url );
+		}
+	}
+
+	render() {
+		return null;
+	}
+}
```

Comment:
To write query components, hooks and the `useDispatch` hook is an awesome tool:
```js
function QueryEmbeds( { siteId, url } ) {
  const dispatch = useDispatch();
  useEffect( () => {
    dispatch( requestEmbed( siteId, url ) );
  }, [ dispatch, siteId, url ] );
  return null;
}
```
And that's the whole component!

Reply:
I like that, thanks - much simpler 👍

Implemented in https://github.com/Automattic/wp-calypso/pull/38608/commits/d26469269e46a7564e12588d612f6489377238dd